src/main/java/frc/robot/subsystems/swerve/Swerve.java

```diff
@@ -54,6 +59,12 @@ public Swerve(SwerveConstants constants, Modules modules, ISwerveGyro gyro) {
         this.gyroInputs = new SwerveGyroInputsAutoLogged();
         this.currentAngleSupplier = this::getAbsoluteHeading;
 
+        this.stateHandler = new SwerveStateHelper(
+                Pose2d::new, //default pose estimator
+                () -> Optional.of(new Translation2d()),
+                this
+        );
```

Comment:
maybe keep it null, and in usage do if null

Reply:
or create one that knows inside to do nothing

Reply:
maybe instead of nothing give the default angle supplier "currentAngleSupplier ", so it can do only aim assist angle

Reply:
it needs the robot position supplier to assist the angle for speaker and stuff so I think living it like this is ok

Reply:
what happen if u use this without setting a new one?
you can give it pose supplier like () -> new Pose2d(0, 0, currenAngleSupplier) and then it can do amp, or do one of the above.